src/Microsoft.ML.StandardTrainers/Standard/MulticlassClassification/MulticlassNaiveBayesTrainer.cs

```diff
@@ -20,14 +20,14 @@
     NaiveBayesTrainer.LoadName,
     NaiveBayesTrainer.ShortName, DocName = "trainer/NaiveBayes.md")]
 
-[assembly: LoadableClass(typeof(MulticlassNaiveBayesModelParameters), null, typeof(SignatureLoadModel),
-    "Multi Class Naive Bayes predictor", MulticlassNaiveBayesModelParameters.LoaderSignature)]
+[assembly: LoadableClass(typeof(NaiveBayesModelParameters), null, typeof(SignatureLoadModel),
+    "Multi Class Naive Bayes predictor", NaiveBayesModelParameters.LoaderSignature)]
 
 [assembly: LoadableClass(typeof(void), typeof(NaiveBayesTrainer), null, typeof(SignatureEntryPointModule), NaiveBayesTrainer.LoadName)]
 
 namespace Microsoft.ML.Trainers
 {
-    public sealed class NaiveBayesTrainer : TrainerEstimatorBase<MulticlassPredictionTransformer<MulticlassNaiveBayesModelParameters>, MulticlassNaiveBayesModelParameters>
+    public sealed class NaiveBayesTrainer : TrainerEstimatorBase<MulticlassPredictionTransformer<NaiveBayesModelParameters>, NaiveBayesModelParameters>
```

Comment:
>NaiveBayesTrainer [](start = 24, length = 17)

Why does `NaiveBayesTrainer` not have `Multiclass` or `MulticlassClassification` in its name?

Reply:
NaiveBayes is a very general probabilistic technique, and could be applied to many different tasks, so I don't understand the rational for removing the task from its name?

---
In reply to: [266106131](https://github.com/dotnet/machinelearning/pull/2968#discussion_r266106131) [](ancestors = 266106131)

Reply:
https://github.com/dotnet/machinelearning/issues/2762#issuecomment-471845715

- {TypeOfTask} is added only only when the algorithm supports multiple kinds of tasks . When added, we will spell out each task fully.

Reply:
We renamed it as per the discussion here.

[https://github.com/dotnet/machinelearning/issues/2762#issuecomment-471845715](https://github.com/dotnet/machinelearning/issues/2762#issuecomment-471845715)

• {TypeOfTask} is added only only when the algorithm supports multiple kinds of tasks . When added, we will spell out each task fully.

Do u have any suggestions ?

---
In reply to: [266107249](https://github.com/dotnet/machinelearning/pull/2968#discussion_r266107249) [](ancestors = 266107249,266106131)

Reply:
My concern is that if we add this algorithm for `binaryclassification` it would require a breaking change to add `Multiclass` to the name `NaiveBayes`. I think that the above guideline works well when the algorithm can only be applied to one learning task. But this algorithm can be applied at least to binary and multiclass so I would suggest to add `MulticlassClassification` to its name, similarly for the `ModelParameters`.

---
In reply to: [266136420](https://github.com/dotnet/machinelearning/pull/2968#discussion_r266136420) [](ancestors = 266136420,266107249,266106131)

Reply:
I see. Currently in ML.NET  we only have NaiveBayes  under the  *Multiclass*  extension of MLContext

So your concern is about a possible breaking change in the future if we want to add NaiveBayes  classfication  as a binary classifier ?    

---
In reply to: [266138507](https://github.com/dotnet/machinelearning/pull/2968#discussion_r266138507) [](ancestors = 266138507,266136420,266107249,266106131)

Reply:
Yes that was my concern, maybe @tfinley@gmail.com  has an opinion about this?

---
In reply to: [266143059](https://github.com/dotnet/machinelearning/pull/2968#discussion_r266143059) [](ancestors = 266143059,266138507,266136420,266107249,266106131)

Reply:
@TomFinley . Would be great to get your thoughts on this.

Reply:
I would keep multiclass on NaiveBayes too. 

---
In reply to: [266143921](https://github.com/dotnet/machinelearning/pull/2968#discussion_r266143921) [](ancestors = 266143921,266143059,266138507,266136420,266107249,266106131)

Reply:
Looks like we favor naming it NaiveBayesMulticlass.  Updating accordingly.

---
In reply to: [266568218](https://github.com/dotnet/machinelearning/pull/2968#discussion_r266568218) [](ancestors = 266568218,266143921,266143059,266138507,266136420,266107249,266106131)